### ISSUE TITLE
Splitchunks plugin naming functionality

### ISSUE THREAD

**Issue by Chrazy** [webpack/webpack#7018]
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
I have two entries and I want to create five chunks in total. Runtime - Vendor - Common - Entry 1 - Entry 2. 

With this config I'm able to produce what I want:
```
optimization: {
	runtimeChunk: {
		name: 'manifest',
	},
	splitChunks: {
		chunks: 'initial',
		cacheGroups: {
			vendor: {
				name: 'vendor',
				chunks: 'all',
				test: /[\\/]node_modules[\\/]/,
			},
			common: {
				name: 'common',
				test: '/src',
				chunks: 'initial',
				minChunks: 2,
			},
		}
	}
}
```
Getting to that config was a lot of trial and error using Bundle analyzer to see what I ended up getting. The only problem I have is the common chunk gets its name from the entry points like: "entry1~
entry2.js" and not "common.js" as I want it to. 

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

**If this is a feature request, what is motivation or use case for changing the behavior?**
I don't know if this is an odd use case or not but it doesn't feel like it's extremely unconventional. 

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**
4.5

**Comment by TrejGun** [comment]
in my case 
```
		splitChunks: {
			cacheGroups: {
				commons: {
					name: "common",
					chunks: "initial",
					minChunks: 2
				}
			}
		}
```
it produces file `0.js` instead of "common.js"

**Comment by sokra** [comment]
`test: '/src',` is wrong. Remove it or use `path.resolve(__dirname, "src")`.

@TrejGun Check if `output.chunkFilename`/`output.filename` contains `[name]`.

**Comment by Chrazy** [comment]
Thanks @sokra , seems like the test was the culprit. I fiddled around with chunks but it's not super clear to me what works and what does not. The one I posted initially and the one below produces the same end result, at least in my case.

So now I get:
- a vendor chunk
- a common chunk (common between entries
- manifest (runtime)
- entries

For others that run in to similar problems, this is what the final config looks like:
`optimization: {
	runtimeChunk: {
		name: 'manifest',
	},
	splitChunks: {
		chunks: 'initial',
		cacheGroups: {
			vendor: {
				name: 'vendor',
				test: /[\\/]node_modules[\\/]/,
			},
			common: {
				name: 'common',
				test: resolve(__dirname, 'src'),
				minChunks: 2,
			},
		}
	}
}`
